solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -331,6 +338,10 @@ public void close() {
           metricManager = null;
         }
       }
+
+      if (rateLimitManager != null) {
```

Comment:
should this be in the finally block instead?

Reply:
Would that not defy the purpose of the destroy method?